browser/ui/brave_shields_data_controller.cc

```diff
@@ -49,14 +59,64 @@ int BraveShieldsDataController::GetTotalBlockedCount() {
           resource_list_blocked_fingerprints_.size());
 }
 
-bool BraveShieldsDataController::GetIsBraveShieldsEnabled() {
+bool BraveShieldsDataController::IsBraveShieldsEnabled() {
   auto* map = HostContentSettingsMapFactory::GetForProfile(
       web_contents()->GetBrowserContext());
   return brave_shields::GetBraveShieldsEnabled(map, GetCurrentSiteURL());
 }
 
 GURL BraveShieldsDataController::GetCurrentSiteURL() {
-  return web_contents()->GetURL();
+  return web_contents()->GetLastCommittedURL();
+}
+
+AdBlockMode BraveShieldsDataController::GetAdBlockMode() {
+  auto* map = HostContentSettingsMapFactory::GetForProfile(
+      web_contents()->GetBrowserContext());
+
+  ControlType control_type_ad =
+      brave_shields::GetAdControlType(map, GetCurrentSiteURL());
+
+  ControlType control_type_cosmetic =
+      brave_shields::GetCosmeticFilteringControlType(map, GetCurrentSiteURL());
+
+  if (control_type_ad == ControlType::ALLOW) {
+    return AdBlockMode::ALLOW;
+  }
+
+  if (control_type_cosmetic == ControlType::BLOCK) {
+    return AdBlockMode::AGGRESSIVE;
+  } else {
+    return AdBlockMode::STANDARD;
+  }
+}
+
+void BraveShieldsDataController::SetAdBlockMode(AdBlockMode mode) {
+  auto* map = HostContentSettingsMapFactory::GetForProfile(
+      web_contents()->GetBrowserContext());
+
+  ControlType control_type_ad;
+  ControlType control_type_cosmetic;
+
+  if (mode == AdBlockMode::ALLOW) {
+    control_type_ad = ControlType::ALLOW;
+  } else {
+    control_type_ad = ControlType::BLOCK;
+  }
+
+  if (mode == AdBlockMode::AGGRESSIVE) {
+    control_type_cosmetic = ControlType::BLOCK;  // aggressive
+  } else {
+    control_type_cosmetic = ControlType::BLOCK_THIRD_PARTY;  // standard
```

Comment:
if AdBlockMode == ALLOW I believe cosmetic filter control should also be ALLOW. cc @antonok-edm to verify

Reply:
after discussing with @antonok-edm we want to set this to ALLOW when AdBlockMode == ALLOW because we CF should only have to check its own setting. Actually enabling that behavior will be done in a follow-up, but this is safe and backwards compatible

Reply:
fyi - this does not impact the logic for `GetAdBlockMode` which is still correct. It's just to avoid CF also having to check the adblock state instead of just checking its own content setting

Reply:
It's been done this way in the past, so it doesn't really matter for this PR - but we should address this (and it will require a settings migration).

I've opened https://github.com/brave/brave-browser/issues/20498 to track that.